src/components/CodeSample/CodeSampleCompile.astro

```diff
@@ -0,0 +1,36 @@
+---
+import CodeSample from "./CodeSample.astro";
+import { tsToJs, getTscOptions } from "./compileTS.ts";
+import { readFile } from "node:fs/promises";
+import CodeSnippet from "../CodeSnippet/CodeSnippet.astro";
+import { Code } from "astro/components";
+
+export interface Props {
+  jsTitle?: string;
+  tsTitle?: string;
+  tsPath: string;
+}
+
+const { tsPath, jsTitle, tsTitle } = Astro.props;
+
+const loadedTs = await readFile(tsPath);
+if (!loadedTs) {
+  throw new Error(`Could not load ${tsPath}`);
+}
+const ts = loadedTs.toString();
+
+const js = await tsToJs(tsPath, getTscOptions());
+---
+
+<CodeSample>
+  {
+    js && (
+      <CodeSnippet slot="js" title={jsTitle} code={js}>
+        <Code code={js} lang="js" />
```

Comment:
It's nice to use `encodeURIComponent()` here so the code uses fewer lines in the `data-code`, also helps avoid errors while copying line breaks, etc.

Considering this component will probably receive both js and jsx components, shouldn't we be using the "jsx" as the `lang` instead? After all (as far as I know) it supports the entire normal JavaScript formatting + JSX.
```suggestion
      <CodeSnippet slot="js" title={jsTitle} code={encodeURIComponent(js)}>
        <Code code={js} lang="jsx" />
```